### ISSUE TITLE
Workflow reviewer cannot upload files to the workflowitems

### ISSUE THREAD

**Issue by marsaoua** [DSpace/DSpace#8133]
**Describe the bug**
While reviewing a workflowitem a reviewer cannot upload a file to the workflowitem. The upload fails. After investigation, it seems, that the UI is sending the wrong request. It sends  `/server/api/submission/workspaceitems/<id>` instead of `/server/api/submission/workflowitems/<id>` hence we are dealing with workflowitems.

**To Reproduce**
Steps to reproduce the behavior (DSpace 7.1):
1. Submit as a simple user an item to a collection with editors role
2. Try to upload a file while editing the workflowitem as reviewer

**Expected behavior**
A workflow reviewer should be able to upload files to workflowitems

**Comment by tdonohue** [comment]
I am able to replicate this on `main` as well. One example way to replicate (using the demo users) is as follows:
1. Login as `dspacedemo+submit@gmail.com` and submit a new Item to the "1-step Workflow collection". Then Logout
2. Now, login as "dspacedemo+admin@gmail.com", go to "MyDSpace -> Workflow tasks" and Claim that newly submitted item to review. Click "Edit" on that item
3. Finally, drag & drop a new file into that Submission. You'll receive an error message & the REST API responds with a `500` response code.

In the `dspace.log` file, the underlying error is a NullPointerException:
```
2022-01-25 20:12:37,297 ERROR unknown 61e93f2c-76f0-4785-ba1b-4739f4fb49d9 org.dspace.app.rest.exception.DSpaceApiExcept
ionControllerAdvice @ An exception has occurred (status:500)
java.lang.NullPointerException: null
        at org.dspace.app.rest.submit.SubmissionService.uploadFileToInprogressSubmission(SubmissionService.java:328) ~[c
lasses/:7.2-SNAPSHOT]
        at org.dspace.app.rest.repository.WorkspaceItemRestRepository.upload(WorkspaceItemRestRepository.java:188) ~[cla
sses/:7.2-SNAPSHOT]
        at org.dspace.app.rest.repository.WorkspaceItemRestRepository.upload(WorkspaceItemRestRepository.java:75) ~[clas
ses/:7.2-SNAPSHOT]
        at org.dspace.app.rest.repository.WorkspaceItemRestRepository$$FastClassBySpringCGLIB$$987ae76d.invoke(<generate
d>) ~[classes/:7.2-SNAPSHOT]
        at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218) ~[spring-core-5.2.5.RELEASE.jar:5.2.
5.RELEASE]
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:771)
 ~[spring-aop-5.2.5.RELEASE.jar:5.2.5.RELEASE]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163) ~[s
pring-aop-5.2.5.RELEASE.jar:5.2.5.RELEASE]
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:749) ~[sprin
g-aop-5.2.5.RELEASE.jar:5.2.5.RELEASE]
        at org.springframework.security.access.intercept.aopalliance.MethodSecurityInterceptor.invoke(MethodSecurityInte
rceptor.java:69) ~[spring-security-core-5.2.2.RELEASE.jar:5.2.2.RELEASE]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[s
pring-aop-5.2.5.RELEASE.jar:5.2.5.RELEASE]
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:749) ~[sprin
g-aop-5.2.5.RELEASE.jar:5.2.5.RELEASE]
        at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:691) ~
[spring-aop-5.2.5.RELEASE.jar:5.2.5.RELEASE]
        at org.dspace.app.rest.repository.WorkspaceItemRestRepository$$EnhancerBySpringCGLIB$$321aae83.upload(<generated
>) ~[classes/:7.2-SNAPSHOT]
        at org.dspace.app.rest.RestResourceController.uploadInternal(RestResourceController.java:594) ~[classes/:7.2-SNA
PSHOT]
        at org.dspace.app.rest.RestResourceController.upload(RestResourceController.java:542) ~[classes/:7.2-SNAPSHOT]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
...
```
(NOTE: I've cut off the full stacktrace to only include the relevant portions above. But, I've checked and there is no "Caused by" portion to the exception.)

@abollini : To me, this appears like it may be a small bug which could be fixed in time for 7.2.  I honestly haven't had the time to dig in further though (as I'm still trying to finish reviews of other PRs).  Would you, or someone on your team, have a chance to glance at this and see if it's easy enough to fix for 7.2?  I'll assign this to you in the hopes you (or someone on your team) can get to it.

**Comment by abollini** [comment]
the issue is on the angular side, the upload component is using the workspaceitem endpoint also when it is used in the workflow
